File: client-end/src/app/models/Apartment.ts

export class Apartment {

    apartmentId?: number = 0
    cityId?: number = 0
    park: boolean = false
    elevator: boolean = false
    numRooms: number = 0
    numBeds: number = 0
    numFloor: number = 0
    porch: boolean = false
    mangal: boolean = false
    accessibleness: boolean = false
    selfDescription: string = ''
    address: string = ''
    lan: number = 0.0//לבדוק איך מגדירים מספר עשרוני
    lat: number = 0.0//לבדוק איך מגדירים מספר עשרוני
    protected_space: boolean = false
    trampoline: Boolean = false
    hammock: Boolean = false
    woodenBench: Boolean = false
    SittingArea: Boolean = false
    isPool: Boolean = false
    priceToBed: number = 0.0
    // מוסיפים לשימוש בצד לקוח
    images?: string[];


    // constructor(selfDescription: string, address: string, numBeds: number,
    //     CostId: number, CityId: number, park: boolean, elevator: boolean, numRooms: number, porch: boolean,
    //     accessibleness: boolean, lan: number, lat: number, numFloor: number,priceToBed:number,
    //     protected_space: boolean, trampoline: Boolean,
    //     hammock: Boolean,
    //     woodenBench: Boolean,
    //     SittingArea: Boolean,
    //     isPool: Boolean = false) {
    //     this.selfDescription = selfDescription
    //     this.numBeds = numBeds
    //     this.CostId = CostId
    //     this.CityId = CityId
    //     this.park = park
    //     this.elevator = elevator
    //     this.numRooms = numRooms
    //     this.numFloor = numFloor
    //     this.porch = porch
    //     this.accessibleness = accessibleness
    //     this.address = address
    //     this.lan = lan
    //     this.lat = lat
    //     this.protected_space = protected_space
    //     this.trampoline = trampoline
    //     this.hammock = hammock
    //     this.woodenBench = woodenBench
    //     this.SittingArea = SittingArea
    //     this.isPool = isPool
    //     this.priceToBed= priceToBed
    // }


}

File: server/models/Apartment.py
from sqlalchemy import Column, Integer, String, Boolean, Float, ForeignKey
from config import Base


class Apartment(Base):
    __tablename__ = "Apartments"
    apartmentId = Column(Integer, primary_key=True, autoincrement=True)
    #  cityId = Column(Integer, ForeignKey("Cities.cityId"))
    cityId = Column(Integer)
    park = Column(Boolean)
    elevator = Column(Boolean)
    numRooms = Column(Integer)
    numBeds = Column(Integer)
    numFloor = Column(Integer)
    porch = Column(Boolean)
    mangal = Column(Boolean)
    accessibleness = Column(Boolean)
    selfDescription = Column(String)
    address = Column(String)
    lat = Column(Float)
    lan = Column(Float)
    protected_space = Column(Boolean)
    trampoline = Column(Boolean)
    hammock = Column(Boolean)
    woodenBench = Column(Boolean)
    SittingArea = Column(Boolean)
    isPool = Column(Boolean)
    priceToBed = Column(Float)
